{
  "repository": {
    "name": "apache/incubator-druid",
    "githubId": 6358188,
    "url": "https://github.com/apache/incubator-druid",
    "isFork": false,
    "original": {
      "name": "",
      "githubId": 0,
      "url": ""
    },
    "isPullRequest": true,
    "pullRequestId": 6129
  },
  "builds": {
    "buggyBuild": {
      "id": 417463468,
      "url": "http://travis-ci.org/apache/incubator-druid/builds/417463468",
      "date": "Aug 19, 2018 4:20:47 AM"
    }
  },
  "commits": {
    "buggyBuild": {
      "repoName": "apache/incubator-druid",
      "branchName": "master",
      "sha": "0a4e3aaf3181662f444047efcc80353600fdf06f",
      "url": "http://github.com/apache/incubator-druid/commit/0a4e3aaf3181662f444047efcc80353600fdf06f",
      "date": "Aug 17, 2018 11:55:00 PM"
    },
    "buggyBuildForkRepo": {
      "repoName": "clintropolis/druid",
      "branchName": "kafka-publish-total-rows",
      "sha": "923537d4090a049f267f96266ec706b5a7cfdf19",
      "url": "https://github.com/clintropolis/druid/commit/923537d4090a049f267f96266ec706b5a7cfdf19",
      "date": "Aug 17, 2018 11:55:00 PM"
    },
    "buggyBuildBaseRepo": {
      "repoName": "apache/incubator-druid",
      "branchName": "master",
      "sha": "62e580050c7e997053a4149a4d5e5f958da7f2b7",
      "url": "https://github.com/apache/incubator-druid/commit/62e580050c7e997053a4149a4d5e5f958da7f2b7",
      "date": "Aug 16, 2018 8:05:45 PM"
    }
  },
  "projectMetrics": {
    "numberModules": 56,
    "numberSourceFiles": 2696,
    "numberTestFiles": 987,
    "numberLibrariesFailingModule": 173
  },
  "tests": {
    "failingModule": "/root/workspace/apache/incubator-druid/417463468/server",
    "overallMetrics": {
      "numberRunning": 85537,
      "numberPassing": 85531,
      "numberFailing": 6,
      "numberErroring": 0,
      "numberSkipping": 88,
      "failures": [
        {
          "failureName": "java.lang.AssertionError",
          "isError": false,
          "occurrences": 6
        }
      ]
    },
    "failingClasses": [
      {
        "testClass": "io.druid.query.lookup.LookupSnapshotTakerTest",
        "numberRunning": 5,
        "numberPassing": 4,
        "numberFailing": 1,
        "numberErroring": 0,
        "numberSkipping": 0
      },
      {
        "testClass": "io.druid.segment.loading.LocalDataSegmentPusherTest",
        "numberRunning": 6,
        "numberPassing": 5,
        "numberFailing": 1,
        "numberErroring": 0,
        "numberSkipping": 0
      },
      {
        "testClass": "io.druid.segment.loading.SegmentLoaderLocalCacheManagerTest",
        "numberRunning": 12,
        "numberPassing": 8,
        "numberFailing": 4,
        "numberErroring": 0,
        "numberSkipping": 0
      }
    ],
    "failureDetails": [
      {
        "testClass": "io.druid.query.lookup.LookupSnapshotTakerTest",
        "testMethod": "testIOExceptionDuringLookupPersist",
        "failureName": "java.lang.AssertionError",
        "detail": "Expected test to throw (an instance of io.druid.java.util.common.ISE and exception with message a string containing \"Exception during serialization of lookups\")",
        "isError": false
      },
      {
        "testClass": "io.druid.segment.loading.SegmentLoaderLocalCacheManagerTest",
        "testMethod": "testRetrySuccessAtSecondLocation[0]",
        "failureName": "java.lang.AssertionError",
        "isError": false
      },
      {
        "testClass": "io.druid.segment.loading.SegmentLoaderLocalCacheManagerTest",
        "testMethod": "testRetryAllFail[1]",
        "failureName": "java.lang.AssertionError",
        "isError": false
      },
      {
        "testClass": "io.druid.segment.loading.SegmentLoaderLocalCacheManagerTest",
        "testMethod": "testRetryAllFail[0]",
        "failureName": "java.lang.AssertionError",
        "isError": false
      },
      {
        "testClass": "io.druid.segment.loading.LocalDataSegmentPusherTest",
        "testMethod": "testPushCannotCreateDirectory",
        "failureName": "java.lang.AssertionError",
        "detail": "Expected test to throw (an instance of java.io.IOException and exception with message a string containing \"Unable to create directory\")",
        "isError": false
      },
      {
        "testClass": "io.druid.segment.loading.SegmentLoaderLocalCacheManagerTest",
        "testMethod": "testRetrySuccessAtSecondLocation[1]",
        "failureName": "java.lang.AssertionError",
        "isError": false
      }
    ]
  },
  "reproductionBuggyBuild": {
    "reproductionDateBeginning": "Aug 19, 2018 4:21:32 AM",
    "reproductionDateEnd": "Aug 19, 2018 5:23:18 AM",
    "totalDuration": 3701,
    "processDurations": {
      "cloning": {
        "nbSteps": 1,
        "totalDuration": 50,
        "stepNames": [
          "CloneRepository"
        ],
        "stepDurations": [
          50
        ]
      },
      "building": {
        "nbSteps": 2,
        "totalDuration": 1139,
        "stepNames": [
          "CheckoutBuggyBuild",
          "BuildProject"
        ],
        "stepDurations": [
          6,
          1133
        ]
      },
      "testing": {
        "nbSteps": 1,
        "totalDuration": 1891,
        "stepNames": [
          "TestProject"
        ],
        "stepDurations": [
          1891
        ]
      }
    },
    "projectRootPomPath": "./workspace/apache/incubator-druid/417463468/pom.xml"
  },
  "patchDiff": {
    "files": {
      "numberAdded": 0,
      "numberChanged": 0,
      "numberDeleted": 0
    },
    "lines": {
      "numberAdded": 0,
      "numberDeleted": 0
    }
  },
  "type": "only_fail"
}